OpenRA.Mods.Common/Traits/Player/ClassicProductionQueue.cs

```diff
@@ -107,11 +107,11 @@ protected override bool BuildUnit(string name)
 
 			foreach (var p in producers.Where(p => !p.Actor.IsDisabled()))
 			{
-				if (p.Trait.Produce(p.Actor, ai, p.Trait.Race))
-				{
-					FinishProduction();
-					return true;
-				}
+				if (!p.Trait.Produce(p.Actor, new[] { ai }, p.Trait.Race))
+					continue;
```

Comment:
I find this change in flow control to be less clear than what it had originally.